src/topology/sheaves/skyscraper.lean

```diff
@@ -0,0 +1,314 @@
+          eq_to_hom_refl, category.comp_id] },
+      { dsimp,
+        split_ifs,
+        rw [←category.assoc, eq_comp_eq_to_hom],
+        exact terminal_is_terminal.hom_ext _ _, }
+    end,
+    λ y (hy : x.is_amalgamation y),
+    begin
+      split_ifs,
+      { generalize_proofs h₁ h₂,
+        have := hy h₂.some h₂.some_spec.1,
+        dsimp at this,
+        split_ifs at this with H, swap,
+        { exact false.elim (H h₂.some_spec.2), },
+        rw [eq_comp_eq_to_hom] at this,
+        rw [this, eq_comp_eq_to_hom, category.assoc, eq_to_hom_trans, eq_to_hom_refl,
+          category.comp_id], },
+      { rw [←eq_comp_eq_to_hom],
+        exact terminal_is_terminal.hom_ext _ _, }
+    end⟩⟩
+
+end
+
+section
+
+-- In this section, we calculate the stalks for skyscraper presheaves.
+-- We need to restrict universe level.
+
+open topological_space
+open category_theory category_theory.limits
+open Top
+open opposite
+
+universes u v
+
+variables {X : Top.{u}} (p₀ : X) {C : Type v} [category.{u} C] (S : C) [has_terminal C]
+
+lemma mem_nhds_of_not_mem_closure_singleton {y : X} (h : ¬p₀ ⤳ y) :
+  ∃ (U : open_nhds y), p₀ ∉ U.1 :=
+begin
+  have := (not_iff_not.mpr specializes_iff_forall_open).mp h,
+  push_neg at this,
+  rcases this with ⟨s, o, h₁, h₂⟩,
+  exact ⟨⟨⟨s, o⟩, h₁⟩, h₂⟩,
+end
+
+variable [Π (U : opens X), decidable (p₀ ∈ U)]
+/--
+The cocone at `S` for the salk functor of `skyscraper_presheaf p₀ S` when `y ∈ closure {p₀}`
+-/
+@[simps] def skyscraper_presheaf_cocone_of_mem_closure₀ {y : X} (h : p₀ ⤳ y) :
+  cocone ((open_nhds.inclusion y).op ⋙ skyscraper_presheaf p₀ S) :=
+{ X := S,
+  ι :=
+  { app := λ U, eq_to_hom $ if_pos (h.mem_open U.unop.1.2 U.unop.2),
+    naturality' := λ U V inc,
+    begin
+      simp only [functor.op_obj, unop_op, functor.comp_map, functor.op_map, skyscraper_presheaf_map,
+        category.id_comp, eq_to_hom_trans, functor.const_obj_map, category.assoc],
+      by_cases hV : p₀ ∈ (open_nhds.inclusion y).obj V.unop,
+      { have hU : p₀ ∈ unop ((open_nhds.inclusion y).op.obj U) := le_of_hom inc.unop hV,
+        split_ifs,
+        erw [eq_to_hom_trans, category.comp_id],
+        refl },
+      { split_ifs with hU;
+        erw [category.comp_id, category.assoc, eq_to_hom_trans, eq_comp_eq_to_hom, eq_to_hom_trans];
+        exact terminal_is_terminal.hom_ext _ _, },
+    end } }
+
+/--
+The canonical map `S ⟶ (skyscraper_presheaf p₀ S t).stalk y` when `y ∈ closure {p₀}`
+-/
+noncomputable def skyscraper_presheaf_from [has_colimits C] {y : X} :
+  S ⟶ (skyscraper_presheaf p₀ S).stalk y :=
+eq_to_hom (skyscraper_presheaf_obj_of_mem S (by tauto : p₀ ∈ ⊤)).symm ≫
+  (skyscraper_presheaf p₀ S).germ (⟨y, trivial⟩ : (⊤ : opens X))
+
+/--
+The cocone at `S` for the salk functor of `skyscraper_presheaf p₀ S` when `y ∈ closure {p₀}` is a
+colimit
+-/
+noncomputable def skyscraper_presheaf_cocone_of_mem_closure₀_is_colimit [has_colimits C]
```

Comment:
It's not clear what the subscript 0 means; how about
```suggestion
noncomputable def skyscraper_presheaf_cocone_is_colimit_of_specializes [has_colimits C]
```
?

Reply:
The subscript was to signify `p₀`, your name is better, I will adopt it through out this file